sites/public/pages/eligibility/age.tsx

```diff
@@ -8,43 +8,61 @@ import {
   FormCard,
   t,
   Form,
-  Field,
   ProgressNav,
+  FieldGroup,
 } from "@bloom-housing/ui-components"
 import FormsLayout from "../../layouts/forms"
 import { useForm } from "react-hook-form"
 import styles from "./EligibilityAge.module.scss"
 import React, { useContext } from "react"
 import { useRouter } from "next/router"
 import { ELIGIBILITY_SECTIONS } from "../../lib/constants"
-import { EligibilityContext } from "../../lib/EligibilityContext"
+import { AgeRangeType, EligibilityContext } from "../../lib/EligibilityContext"
 import FormBackLink from "../../src/forms/applications/FormBackLink"
 import { eligibilityRoute } from "../../lib/helpers"
 
 const EligibilityAge = () => {
   const router = useRouter()
-  // Check if they need to be 18 or older to apply?
-  const MIN_AGE = 0
-  const MAX_AGE = 120
   const CURRENT_PAGE = 2
 
   /* Form Handler */
   // eslint-disable-next-line @typescript-eslint/unbound-method
-  const { handleSubmit, register, errors, setError } = useForm()
+  const { handleSubmit, register } = useForm()
   const { eligibilityRequirements } = useContext(EligibilityContext)
 
   const onSubmit = (data) => {
-    if (isAgeValid(data.age)) {
+    if (data.age !== "preferNotSay") {
       eligibilityRequirements.setAge(data.age)
-      void router.push(eligibilityRoute(CURRENT_PAGE + 1))
-    } else {
-      setError("age", { type: "manual", message: "" })
     }
+    void router.push(eligibilityRoute(CURRENT_PAGE + 1))
```

Comment:
I know this pre-dates your change, but my instinct is that we want to `await router.push(...)` and make `onSubmit` async, unless you have a sense of why this is better?

Reply:
I guess I might as well update all of them for consistency.